### DISCUSSION TITLE
How to change the keyboard layout for encryption passphrase entry?

### DISCUSSION THREAD

**Discussion by phlibi** [zbm-dev/zfsbootmenu#622] [Questions] [answered]
Hi everyone,

some searching around brought me nowhere, so I finally ask here: How can I select the keyboard layout that is used when I have to enter the encryption passphrase for the zpool? It seems to default to US English, which is certainly a reasonable default, but it would simplify typing for me if I could change that.
In case there is actually no configuration option in the prebuilt image for this, then what would be the easiest way to configure it in a custom image?

Thanks!

Best regards,
Philipp

**Comment by classabbyamp** [comment, marked as answer]
`rd.vconsole.keymap=...` in zfsbootmenu's cmdline should work

**Comment by phlibi** [comment]
That was quick, thanks! Can I pass this command line somehow in `efibootmgr` when using the stock bundled EFI image of ZFSBootMenu? Or do I need to use the separate loader and initrd to do this or build a custom image? I'm not really used to dealing with EFI boot entries yet, having used Grub all the time before. I see there is a `--unicode` option in `efibootmgr` to "pass extra args", but I don't know if these are of any use with the bundled image.

**Reply by classabbyamp** [reply]
not all efi firmwares support that. the zfsbootmenu efi has an embedded commandline in it that you can edit with the [zbm-kcl script](https://get.zfsboot.menu/zbm-kcl)

**Reply by phlibi** [reply]
Perfect, thanks! Works like a charm. As reference for future readers, here's what I did:

```
$ wget https://raw.githubusercontent.com/zbm-dev/zfsbootmenu/master/bin/zbm-kcl
$ sudo bash zbm-kcl -a rd.vconsole.keymap=de_CH-latin1 /boot/efi/EFI/ZBM/VMLINUZ.EFI
```

**Reply by numerfolt** [reply]
Edit:
I found out that when using rEFInd I needed to add the `rd.vconsole.keymap=de-latin1` to the corresponding entry in the refind.conf in /boot/efi/EFI/refind/.
This solved it for me.


--- old post ---
Please excuse my necroing:
I changed my keymap like this `bash zbm-kcl -a rd.vconsole.keymap=de-latin1 /boot/efi/EFI/ZBM/VMLINUZ.EFI` and it actually shows up in the zbm like this.
But nevertheless, I need to enter my password using the US keyboard layout.

How would I troubleshoot why it isnt working?

**Comment by mdPlusPlus** [comment]
I'm not using rEFInd, neither `zfs set org.zfsbootmenu:commandline="quiet kbd=de init=/lib/systemd/systemd"`, nor `zfs set org.zfsbootmenu:commandline="quiet rd.vconsole.keymap=de init=/lib/systemd/systemd"` seem to work. Initially I have to use the US keymap.

Issuing `zbm-kcl -a rd.vconsole.keymap=de-latin1 /boot/efi/EFI/ZBM/VMLINUZ.EFI` however seems to work.  
Is there a way to incorporate that into a org.zfsbootmenu attribute?
